### PR TITLE
C: Implement Tuple/List Compare

### PR DESCRIPTION
**Pull request by Smit-create**

### CODE REVIEW

**Comment by Smit-create** [comment]
@certik @czgdp1807 This is ready for review.

File: src/libasr/codegen/c_utils.h [hunk 1]
```diff
@@ -906,7 +912,6 @@ class CCPPDSUtils {
             tmp_gen += indent + tab + "idx1 = step < 0 ? (idx1 >= s_len ? s_len-1 : idx1) : idx1;\n";
             tmp_gen += indent + tab + list_struct_type + " *__tmp = (" +
                     list_struct_type + "*) malloc(sizeof(" + list_struct_type + "));\n";
-            // tmp_gen += indent + tab + list_struct_type + " __tmp;\n";
```
Comment (czgdp1807) [inline src/libasr/codegen/c_utils.h:909]: Should be removed?

Comment (Smit-create) [reply src/libasr/codegen/c_utils.h:909]: Yep.

Comment (czgdp1807) [reply src/libasr/codegen/c_utils.h:909]: Remove it and then I will approve.

Comment (Smit-create) [reply src/libasr/codegen/c_utils.h:909]: That comment is unnecessary and we can remove it

Comment (czgdp1807) [reply src/libasr/codegen/c_utils.h:909]: Have you pushed the changes? Its still showing up in the diff on Github.

**Review by czgdp1807 — APPROVED** [review]

**Pull request merged**